tools/flagd-http-connector/src/main/java/dev/openfeature/contrib/tools/flagd/resolver/process/storage/connector/sync/http/HttpConnector.java

```diff
@@ -0,0 +1,227 @@
+    private PayloadCache payloadCache;
+    private HttpCacheFetcher httpCacheFetcher;
+    private int payloadCachePollTtlSeconds;
+    private boolean usePollingCache;
+
+    @NonNull private String url;
+
+    private URI uri;
+
+    /**
+     * HttpConnector constructor.
+     *
+     * @param httpConnectorOptions options for configuring the HttpConnector.
+     */
+    @Builder
+    public HttpConnector(HttpConnectorOptions httpConnectorOptions) {
+        this.pollIntervalSeconds = httpConnectorOptions.getPollIntervalSeconds();
+        this.requestTimeoutSeconds = httpConnectorOptions.getRequestTimeoutSeconds();
+        ProxySelector proxySelector = NO_PROXY;
+        if (httpConnectorOptions.getProxyHost() != null && httpConnectorOptions.getProxyPort() != null) {
+            proxySelector = ProxySelector.of(
+                    new InetSocketAddress(httpConnectorOptions.getProxyHost(), httpConnectorOptions.getProxyPort()));
+        }
+        this.url = httpConnectorOptions.getUrl();
+        uri = URI.create(url);
+        this.headers = httpConnectorOptions.getHeaders();
+        this.httpClientExecutor = httpConnectorOptions.getHttpClientExecutor();
+        scheduler = Executors.newScheduledThreadPool(httpConnectorOptions.getScheduledThreadPoolSize());
+        this.client = HttpClient.newBuilder()
+                .connectTimeout(Duration.ofSeconds(httpConnectorOptions.getConnectTimeoutSeconds()))
+                .proxy(proxySelector)
+                .executor(this.httpClientExecutor)
+                .build();
+        this.queue = new LinkedBlockingQueue<>(httpConnectorOptions.getLinkedBlockingQueueCapacity());
+        this.payloadCache = httpConnectorOptions.getPayloadCache();
+        if (payloadCache != null && Boolean.TRUE.equals(httpConnectorOptions.getUseFailsafeCache())) {
+            this.failSafeCache = FailSafeCache.builder()
+                    .payloadCache(payloadCache)
+                    .payloadCacheOptions(httpConnectorOptions.getPayloadCacheOptions())
+                    .build();
+        }
+        if (Boolean.TRUE.equals(httpConnectorOptions.getUseHttpCache())) {
+            httpCacheFetcher = new HttpCacheFetcher();
+        }
+        payloadCachePollTtlSeconds = pollIntervalSeconds; // safety margin
+        this.usePollingCache = Boolean.TRUE.equals(httpConnectorOptions.getUsePollingCache());
+    }
+
+    @Override
+    public void init() throws Exception {
+        log.info("init Http Connector");
+    }
+
+    @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "parent defines the interface")
+    @Override
+    public BlockingQueue<QueuePayload> getStreamQueue() {
+        boolean success = fetchAndUpdate();
+        if (!success) {
+            log.info("failed initial fetch");
+            updateFromCache();
+        }
+        Runnable pollTask = buildPollTask();
+        scheduler.scheduleWithFixedDelay(pollTask, pollIntervalSeconds, pollIntervalSeconds, TimeUnit.SECONDS);
```

Comment:
Unless I'm missing something, I think you should do this in init. As written, aren't we possibly scheduling multiple tasks with this every time getStreamQueue is called? I think it's better to do this just once in init and then we get the benefit of the framework making sure this doesn't happen more than once.

Reply:
Well, intuitively it is right, and it is confusing.
By what I observed by the call hierarchy, the only place calling it is indeed FlagdProvider.initialize(), so it looks to me that this is by the design of the _QueueSource_.
![image](https://github.com/user-attachments/assets/dfefa55f-193c-404d-8bc3-67fb0550dfaf)